### ISSUE TITLE
Implement reporting mockup

### ISSUE THREAD

**Issue by awong-dev** [department-of-veterans-affairs/SCIInterim#33]
As per #18

**Comment by awong-dev** [comment]
I didn't get a chance to merge this down and look at it, but skimming the code, I think your proposal is to expose the reports via JSON in an API?

Since most of this app is not AJAX, I'm uncertain if we would want to add that mode of complexity in yet. If we kicked it old-skool Web 1.0 with full page refreshes per request, would it be more complicated?

Will look at it more tomorrow...

**Comment by huttj** [comment]
Actually, the API part was because I wanted to do entity selection without a page load. If you look at [pick_entity.slim:26](https://github.com/awong-dev/SCIInterim/blob/reports/app/views/reports/pick_entity.slim#L26) you'll see that I added a script to load users based on the search form.

I commented that out and switch back to the current page-refresh method, though. So, the API controller is unnecessary now (except for populating the DB, but I'm sure there's a much better way to do that--I just didn't have the time to figure it out).

Right now, I am struggling to figure out the best way to generate/display the reports.

It looks like each report will need a method that collects the required data, as well as a view that can be rendered with the data and included on the run report page. I was hoping to make them a little more self-contained, but I can't convince myself to put HTML in the code.

So, I think the next step is to separate the report definitions into a separate set of classes that extend from a base "report" class and a set of corresponding views, and just load them dynamically into report list controller.

Then, each report could be responsible for rendering itself and that can be included into the view-report page.

Any of that make sense? :smile:
